sound/soc/sof/intel/hda-dsp.c

```diff
@@ -446,26 +446,38 @@ static int hda_dsp_set_D0_state(struct snd_sof_dev *sdev,
 		    !hda_enable_trace_D0I3_S0 ||
 		    sdev->system_suspend_target != SOF_SUSPEND_NONE)
 			flags = HDA_PM_NO_DMA_TRACE;
+
+		ret = hda_dsp_send_pm_gate_ipc(sdev, flags);
+		if (ret < 0) {
+			dev_err(sdev->dev,
+				"error: PM_GATE ipc error %d\n", ret);
+			return -EIO;
```

Comment:
Is there a reason to change the return code?

Reply:
yes, good to keep it. Thanks!